include/zephyr/drivers/flash.h

```diff
@@ -128,6 +148,11 @@ typedef int (*flash_api_read_jedec_id)(const struct device *dev, uint8_t *id);
 typedef int (*flash_api_ex_op)(const struct device *dev, uint16_t code,
 			       const uintptr_t in, void *out);
 
+struct flash_api_infoword {
```

Comment:
I'm okay with adding this infoword, but if we are extending the API, we should add all required information here, like 	write_block_size and erase_value, then simply replace the flash_get_parameters() to intentionally break all existing drivers, since they will all need to now signal that they support and require erase.

Reply:
Accessing infoword by callback may be OK, but will require, probably, caching by user as calls are more expensive than direct memory access (i.e. checking bits in infoword).
That is why macros work better here, because they basically access the bits.

Reply:
Macros are just less scalable since a single subsystem will likely interact with different "flash" devices with different properties, its adding many pathways to the code, compared to reading the info structure

Reply:
Contracts for embeded devices are scaled by bucks you save for not paying for extra flash/storage, because this will be biggest area on die.

Reply:
All products I have worked with have both an internal and external flash, these are pretty much always different in either write size, and pretty much always erase page layout. I see the scenario where there is no "flash" device which requires erase being an outlier